internal/validations/observability/common/validate.go

```diff
@@ -0,0 +1,52 @@
+package common
+
+import (
+	"fmt"
+	obsv1 "github.com/openshift/cluster-logging-operator/api/observability/v1"
+	internalobs "github.com/openshift/cluster-logging-operator/internal/api/observability"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func ValidateConfigMapOrSecretKey(name string, keys []*obsv1.ConfigMapOrSecretKey, secrets map[string]*corev1.Secret, configMaps map[string]*corev1.ConfigMap) (conditions []metav1.Condition) {
+	if len(keys) == 0 {
+		return nil
+	}
+	for _, key := range keys {
```

Comment:
nit: can we find other word instead of `key` to avoid `key.Key`, e.g: `tag`, `reference`

Reply:
how about "configs" instead of keys   and "entry" instead of "key".  This is set of config with multiple entries in the set